src/main/java/de/refactoringBot/refactoring/supportedRefactorings/RenameMethod.java

```diff
@@ -126,43 +148,189 @@ public String performRefactoring(BotIssue issue, GitConfiguration gitConfig) thr
+		if (oldMethodName != null && oldMethodName.equals(issue.getRefactorString())) {
+			throw new BotRefactoringException("New method name must differ from the current one!");
+		}
 
-		// Add class to the TO-DO list
-		allRefactorings.addToDoClass(methodClassSignature);
+		// Add all Subclasses and their Superclasses to AST-Tree
+		while (true) {
+			int before = refactoring.getClasses().size();
+			refactoring = addSubClasses(refactoring, allJavaFiles);
+			int after = refactoring.getClasses().size();
+			// Break if all classes found
+			if (before == after) {
+				break;
+			}
+		}
 
-		// Create super tree recursively
-		allRefactorings = getSuperTree(allRefactorings, allJavaFiles, issueFilePath);
+		// Find all Methods and Method-Calls for renaming
+		refactoring = findMethods(refactoring, allJavaFiles, localMethodSignature);
+		refactoring = findMethodCalls(refactoring, allJavaFiles);
 
-		// Add sub tree recursively
-		while (true) {
+		// Rename method declarations and their calls
+		renameFindings(refactoring, issue.getRefactorString());
 
-			// Count classes before adding subclasses
-			int beforeSubtree = allRefactorings.getDoneClasses().size();
+		return "Renamed method '" + oldMethodName + "' to '" + issue.getRefactorString() + "'."
+				+ refactoring.getWarning();
+	}
 
-			// Add subclasses
-			for (String javaFile : allJavaFiles) {
-				allRefactorings = addSubTree(allRefactorings, allJavaFiles, javaFile);
-			}
+	/**
+	 * This method scanns all java files for classes that could be subclasses in our
+	 * AST-Tree.
+	 * 
+	 * @param refactoring
+	 * @param allJavaFiles
+	 * @return refactoring
+	 * @throws FileNotFoundException
+	 * @throws BotRefactoringException
+	 */
+	private ParserRefactoring addSubClasses(ParserRefactoring refactoring, List<String> allJavaFiles)
+			throws FileNotFoundException {
 
-			// Count classes after adding subclasses
-			int afterSubtree = allRefactorings.getDoneClasses().size();
+		// Search all Java-Files
+		for (String javaFile : allJavaFiles) {
+			// parse a file
+			FileInputStream filepath = new FileInputStream(javaFile);
+			CompilationUnit compilationUnit = LexicalPreservingPrinter.setup(JavaParser.parse(filepath));
 
-			// If no more new subclasses found
-			if (beforeSubtree == afterSubtree) {
-				break;
+			// Get all Classes
+			List<ClassOrInterfaceDeclaration> classes = compilationUnit.findAll(ClassOrInterfaceDeclaration.class);
+
+			// Search all Classes
+			for (ClassOrInterfaceDeclaration currentClass : classes) {
+				List<String> classAncestors = new ArrayList<String>();
+				boolean isSubClass = false;
+				boolean hasExternalDep = false;
+
+				// Get all Super-Classes
+				List<ResolvedReferenceType> ancestors = null;
+				try {
+					ancestors = currentClass.resolve().getAllAncestors();
+				} catch (InvalidPathException i) {
+					ancestors = getAllAncestors(currentClass.resolve());
+					hasExternalDep = true;
+				} catch (Exception e) {
+					continue;
```

Comment:
At least log the exception message at debug level

Reply:
Added logs at debug level.